stores/artworks.ts

```diff
@@ -1,47 +1,110 @@
 import { defineStore } from 'pinia'
+import { useCategoryStore } from "@/stores/categories";
 
-interface State {
-  artworks: Artwork[]
-}
-
-export const useArtworkStore = defineStore('artworks', {
-  state: (): State => ({
-    artworks: []
-  }),
-  getters: {
-    countByCategory: (state) => {
-      return (categoryId: number) => state.artworks.filter((art: Artwork) => art.categoryId === categoryId).length;
-    },
-  },
-  actions: {
-    upsertArtwork (artwork: Artwork, categoryId?: number) {
-      // If we passed a categoryId, add it to the artwork
-      if (categoryId >= 0) {
-        artwork.categoryId = categoryId;
-      }
+const defaultPageSize = 16;
 
-      // See if the artwork already is in the collection
-      const idx = this.artworks.findIndex((art: Artwork) => {
-        return art.categoryId === categoryId && art.id === artwork.id
-      });
+export const useArtworkStore = defineStore('artworks', () => {
+  const artworks = ref([]);
+  const totalArtworks = computed(() => artworks.value.length);
 
-      // If the artwork isn't in there add it, else update it.
-      if (idx === -1) {
-        this.artworks.push(artwork);
-      }
-      else {
-        this.artworks.splice(idx, 1, artwork);
+  function findById (id: string, categoryId: string): Artwork | undefined {
```

Comment:
Waarom heeft deze ook een category ID nodig? Zijn artwork IDs niet uniek?

Reply:
Aangezien de id's nu de eigenlijke slugs zijn, gebruik ik de category als extra check om zeker te zijn dat we de unique heb.